make/buf/scripts/npmpublish.bash

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+# "-u" can be omitted if you pass arguments to your bash script, as dealing with 
+# "-u" and "${1}", "${2}", etc is very annoying
+set -euo pipefail
+
+# ../.. down to the root of the repository
+# this can be ommitted for truly universal scripts
+DIR="$(CDPATH= cd "$(dirname "${0}")/../../.." && pwd)"
+cd "${DIR}"
+
+
+if [[ -z "${VERSION}" ]]; then
+    echo "Must provide VERSION in environment to publish" 1>&2
+    exit 1
+fi
+
+# Allow an additional version to let us bump the version published to npm.
+
+NPM_VERSION="${VERSION}"
+
+if [ -n "${REVISION-}" ]; then
+	NPM_VERSION="${VERSION}-${REVISION}"
+fi
+
+update_npm_platform_files() {
+	mkdir -p "npm/@bufbuild/buf-${2}/bin"
+
+	if [[ ${2} =~ "win32" ]]
+	then
+		# Download the binaries
+		curl -Lf "https://github.com/bufbuild/buf/releases/download/v${VERSION}/buf-${1}.exe" -o "npm/@bufbuild/buf-${2}/bin/buf.exe"
```

Comment:
If I understand correctly, we have to make a regular release first, then run this as a follow-up to publish to npm, right? Seems okay to me for a start, just wanted to understand.

Reply:
Yes, this is most useful when triggered from a release in the `npm-publish.yaml` github action. I wanted to split this process out from the rest of the release to keep things loose and flexible.